app/models/piece.rb

```diff
@@ -1,3 +1,60 @@
 class Piece < ApplicationRecord
   belongs_to :game
+
+  # customize this method for each subclass (bishop, pawn, etc), using only what is needed for that piece's movement
+  # def obstructed?(destination_x, destination_y)
+  #  starting_point = [self.x_position, self.y_position]
+  #  ending_point = [destination_x, destination_y]
+  #   if horizontal_or_vertical_obstruction?(starting_point, ending_point) || diagonal_obstruction?(starting_point, ending_point) || invalid?(starting_point, ending_point)
+  #     return true
+  #   else
+  #     return false
+  #   end
+  # end
```

Comment:
This same method will be used for almost all of the pieces. We won't be overriding it for each type of piece.

We will be writing another method for each piece (called `valid_move?`). `obstructed` will only be in `Piece` though.